### PR TITLE
Adds saveUninitialized option

### PR DESCRIPTION
**Pull request by Green-Cat**
Implements feature discussed #36.
The option defaults to true to maintain backward compatibility.
Please let me know if you'd like me to change/fix anything.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,6 +69,9 @@ Compatible to stores from [express-session](https://github.com/expressjs/session
 Defaults to a simple in memory store.</br>
 **Note**: The default store should not be used in a production environment because it will leak memory.
 
+##### saveUninitialized (optional) 
+Should the session be persisted, even if no data is saved into it? Defaults to `true`.
```
Comment (SerayaEryn) [inline README.md]: Could you please reword this not to be a question?

**Comment by SerayaEryn** [comment]
Thank you for opening the PR.

I see a problem in the way how the `sessionHash` is stored. It shares the `sessionHash` between different requests.

If we have a request it will set the `sessionHash` to the hash of the session from the request. Now if we have a second request before the first request is finished, it will change the `sessionHash` and the session of the first request will be saved - even if it shouldn't be be saved.

I wrote a test to reproduce the problem: https://gist.github.com/SerayaEryn/27847ce1e89e310303e4b0e5edb03444
An alternative would be to decorate the `request` with the `sessionHash`.

But I wonder if it is even necessary to calculate the hash to detect a new but not modified session. 
A new session has 4 properties. If something is added to the session the number of properties will change.
Shouldn't something like the following be enough? It would be more performant, too.
```js
Object.keys(session).length !== 4
```

**Comment by Green-Cat** [comment]
Well spotted, I completely missed that the sessionHash wasn't per request.
I switched to the logic you suggested above, I agree it is much better.
Thanks for reviewing the PR, let me know if you'd like any further changes.

**Comment by SerayaEryn** [comment]
Looks good to me now.

**Pull request merged**